File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        //1  2  4  6 nums
        //1  1  2  8 prefix
        //48 24 6  1 suffix
        int[] prefix = new int[nums.length];
        int[] suffix = new int[nums.length];
        int product = 1;
        for(int i = 0; i < nums.length; i++){
            prefix[i] = product;
            product = product * nums[i];
        }
        product = 1;
        for(int i = nums.length-1; i >= 0; i--){
            suffix[i] = product;
            product = product * nums[i];
        }
        for(int i = 0; i < nums.length; i++){
            nums[i] = prefix[i] * suffix[i];
        }
        return nums;
    }
}  

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1


        while l <= r:
            if nums[l] <= nums[r]:
                return nums[l]
            m = math.floor((r+l) / 2)
            if nums[m] >= nums[l] and nums[m] >= nums[r]:
                l = m + 1
            elif nums[m] <= nums[l] and nums[m] <= nums[r]:
                r = m 

        return -1